lega/verify.py

```diff
@@ -19,102 +19,97 @@
-
-        context.verify_mode = ssl.CERT_NONE
-        # Require server verification
-        if CONF.get_value('quality_control', 'verify_peer', conv=bool, default=False):
-            LOG.debug("Require server verification")
-            context.verify_mode = ssl.CERT_REQUIRED
-            cacertfile = CONF.get_value('quality_control', 'cacertfile', default=None)
-            if cacertfile:
-                context.load_verify_locations(cafile=cacertfile)
-
-        # Check the server's hostname
-        server_hostname = CONF.get_value('quality_control', 'server_hostname', default=None)
-        verify_hostname = CONF.get_value('quality_control', 'verify_hostname', conv=bool, default=False)
-        if verify_hostname:
-            LOG.debug("Require hostname verification")
-            assert server_hostname, "server_hostname must be set if verify_hostname is"
-            context.check_hostname = True
-            context.verify_mode = ssl.CERT_REQUIRED
-
-        # If client verification is required
-        certfile = CONF.get_value('quality_control', 'certfile', default=None)
-        if certfile:
-            LOG.debug("Prepare for client verification")
-            keyfile = CONF.get_value('quality_control', 'keyfile')
-            context.load_cert_chain(certfile, keyfile=keyfile)
-
-    try:
-        with urlopen(keyurl, context=context) as response:
-            assert(response.status == 200)
-            privkey = response.read()
-            if not privkey:  # Correcting a bug in the EGA keyserver
-                # When key not found, it returns a 200 and an empty payload.
-                # It should probably be changed to a 404
-                raise exceptions.PGPKeyError('No PGP key found')
-            return header_to_records(privkey, header, os.environ['LEGA_PASSWORD']), keyid
-    except HTTPError as e:
-        LOG.error(e)
-        msg = str(e)
-        if e.code == 404:  # If key not found, then probably wrong key.
-            raise exceptions.PGPKeyError(msg)
-        # Otherwise
-        raise exceptions.KeyserverError(msg)
-    # except Exception as e:
-    #     raise exceptions.KeyserverError(str(e))
+class ChecksumFile():
+    '''Fake IO writer, accepting bytes to checksum but not writing them anywhere'''
+    def __init__(self):
+        self.md = hashlib.sha256()
+
+    def write(self, data):
+        self.md.update(data)
+
+    def hexdigest(self):
+        return self.md.hexdigest()
+
+class PrependHeaderFile():
+    def __init__(self, header, bulk):
+        assert(header)
+        self.header = header
+        self.file = bulk
+        self.pos = 0
+        self.header_length = len(header)
+
+    def seek(self, offset, whence):
+        # Not needed because we decrypt all of it, and not only a range
+        raise NotImplementedError(f'Moving file pointer to {offset}: Unused case')
+
+    def read(self, size=-1):
+        # if size < 0: # read all
```

Comment:
do we need it, or is there a reason this is commented out? - I assume debugging

Reply:
I want to raise an exception in case it is called. And nope, we do not `seek` when decrypting the whole file, but there are other cases where one can `seek` forward or backwards. I kept it there to find the error when it occurs instead of digging.